README.md

```diff
@@ -223,9 +224,234 @@ object Test extends App {
+  .setInt32Value(Int32Value.newBuilder().setValue(123))
+  .setLongValue(Int64Value.newBuilder().setValue(456))
+  .setDuration(GpbDuration.newBuilder().setSeconds(123).setNanos(456))
+  .setTimestamp(GpbTimestamp.newBuilder().setSeconds(123).setNanos(456))
+  .setListValue(ListValue.newBuilder().addValues(Value.newBuilder().setNumberValue(456.789)))
+  .setStruct(Struct.newBuilder().putFields("mapKey", Value.newBuilder().setNumberValue(42).build()))
+  .build()
+
+val expected = CaseClassExtensions(
+  boolValue = BoolValue.newBuilder().setValue(true).build(),
+  int32Value = Int32Value.newBuilder().setValue(123).build(),
+  longValue = Int64Value.newBuilder().setValue(456).build(),
+  duration = GpbDuration.newBuilder().setSeconds(123).setNanos(456).build(),
+  timestamp = GpbTimestamp.newBuilder().setSeconds(123).setNanos(456).build(),
+  listValue = ListValue.newBuilder().addValues(Value.newBuilder().setNumberValue(456.789)).build(),
+  struct = Map("mapKey" -> NumberValue(42))
+)
+
+val Good(converted) = gpb.asCaseClass[CaseClassExtensions]
+
+assertResult(expected)(converted)
+
+assertResult(Good(gpb))(converted.asGpb[ExtensionsMessage])
+
+```
+
+Conversion to basic Scala types is supported too:
+
+```scala
+import com.avast.cactus._
+import com.avast.cactus.v3.ValueOneOf._
+import com.google.protobuf.{BoolValue, BytesValue, DoubleValue, FloatValue, Int32Value, Int64Value, ListValue, StringValue, Struct, Value, Duration => GpbDuration, Timestamp => GpbTimestamp}
+
+val gpb = ExtensionsMessage.newBuilder()
+  .setBoolValue(BoolValue.newBuilder().setValue(true))
+  .setInt32Value(Int32Value.newBuilder().setValue(123))
+  .setLongValue(Int64Value.newBuilder().setValue(456))
+  .setDuration(GpbDuration.newBuilder().setSeconds(123).setNanos(456))
+  .setTimestamp(GpbTimestamp.newBuilder().setSeconds(123).setNanos(456))
+  .setListValue(ListValue.newBuilder().addValues(Value.newBuilder().setNumberValue(456.789)))
+  .setStruct(Struct.newBuilder().putFields("mapKey", Value.newBuilder().setNumberValue(42).build()))
+  .build()
+
+case class CaseClassExtensionsScala(boolValue: Boolean,
+                                    int32Value: Int,
+                                    longValue: Option[Long], // Option is supported
+                                    listValue: Seq[ValueOneOf],
+                                    duration: Duration,
+                                    timestamp: Instant,
+                                    struct: Map[String, ValueOneOf]
+                                    )
+                                    
+gpb.asCaseClass[CaseClassExtensionsScala]
+```
 
 ## Appendix
 
-The library is compiled with dependency on GPB 2.6.1. It should be safe to exclude the dependency and use version
-`2.5.x` or `2.4.x` instead since it uses the `MessageLite` interface only as a "marker trait".
+The library is compiled with optional dependency on GPB 3.3.0. It should be safe to use the library with versions `2.6.x`, `2.5.x` or 
```

Comment:
Because I tried it ;-) The functionality doesn't depend on real content of the `MessageLite` interface. It's used just as context bound in package object to provide the implicit converter. Everything else is then done by macros which are tested separately for v2 and v3. BTW, surprisingly, class generated for GPB 2.6 works with GPB 3.3 dependency. They made new interfaces like `com.google.protobuf.GeneratedMessageV3`.